File: src/geodecoder/geojson.py
from typing import Literal

from pydantic import BaseModel, Field


class Point(BaseModel):
    type: Literal["Point"] = "Point"
    coordinates: tuple[str, str]


Geometry = Point


class Properties(BaseModel):
    description: str
    iconCaption: str
    marker_color: str = Field(..., alias="marker-color")

    class Config:
        allow_population_by_field_name = True


class Feature(BaseModel):
    type: Literal["Feature"] = "Feature"
    id: int
    geometry: Geometry
    properties: Properties


class FeatureCollection(BaseModel):
    type: Literal["FeatureCollection"] = "FeatureCollection"
    features: list[Feature]

File: src/lib/geojson.ts
import { z } from 'zod'

const pointSchema = z.object({
  type: z.literal('Point'),
  coordinates: z.tuple([z.number(), z.number()])
})

const propertiesSchema = z.object({
  description: z.string(),
  iconCaption: z.string(),
  'marker-color': z.string()
})

const featureSchema = z.object({
  type: z.literal('Feature'),
  id: z.number(),
  geometry: pointSchema,
  properties: propertiesSchema
})

export type Feature = typeof featureSchema['_output']

export const featureCollectionSchema = z.object({
  type: z.literal('FeatureCollection'),
  features: z.array(featureSchema)
})

export type FeatureCollection = typeof featureCollectionSchema['_output']
